hie-plugin-api/Haskell/Ide/Engine/PluginDescriptor.hs

```diff
@@ -214,7 +233,8 @@ data ParamVal (t :: ParamType) where
 class ValidResponse a where
   jsWrite :: a -> Object -- ^ Serialize to JSON Object
   jsRead  :: Object -> Parser a -- ^ Read from JSON Object
-
+  responseTypeName :: a -> T.Text -- ^ Type name in command description.
```

Comment:
The command return type needs to be a ValidResponse anyway, so if we use TypeRep there'll be an exact correspondance between that string and the response structure the IDE can expect. I'll change to use the TypeRep.